tests/test_generator.py

```diff
@@ -0,0 +1,658 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import io
+import logging
+import pytest
+
+from pymeasure.adapters import ProtocolAdapter
+from pymeasure.instruments import Channel, Instrument
+from pymeasure.instruments.hcp import TC038, TC038D
+
+from generator import (write_test, write_parametrized_test, write_parametrized_method_test,
+                       parse_stream, Generator, ByteStreamHandler)
+
+
+class FakeChildChannel(Channel):
+
+    channel_control = Channel.control("G{{ch}}.{ch}", "S{{ch}}{ch} %f",
+                                      "Control something. (float)")
+
+
+class FakeChannel(Channel):
+
+    port = Channel.ChannelCreator(FakeChildChannel, id=1)
+
+    channel_control = Channel.control("G{ch}", "S{ch} %f", "Control something. (float)")
+
+    def funny_method(self, value=7):
+        """Some method for testing purposes."""
+        return float(self.ask("M{ch} " + str(value)))
+
+
+class FakeInstrument(Instrument):
+
+    ch_A = Instrument.ChannelCreator(FakeChannel, id="A")
+
+    def __init__(self, adapter, name="Fake"):
+        super().__init__(adapter, name)
+
+    i_control = Instrument.control("G", "S %f", "Control instrument. (float)")
+
+
+@pytest.fixture
+def file():
```

Comment:
Smart to avoid touching the file system (which is quite slow)!

Reply:
Thanks